src/Scripting/Core/ScriptOptions.cs

```diff
@@ -182,193 +183,50 @@ private ScriptOptions(ScriptOptions other)
         /// </summary>
-        public ScriptOptions WithCustomMetadataResolution(MetadataReferenceResolver resolver) =>
+        public ScriptOptions WithMetadataResolver(MetadataReferenceResolver resolver) =>
             MetadataResolver == resolver ? this : new ScriptOptions(this) { MetadataResolver = resolver };
 
         /// <summary>
-        /// Creates a new <see cref="ScriptOptions"/> with <see cref="SourceResolver"/> set to the default source resolver for the current platform.
-        /// </summary>
-        /// <param name="searchPaths">Directories to be used by the default resolver when resolving script file names.</param>
-        /// <remarks>
-        /// The default resolver looks up scripts in specified <paramref name="searchPaths"/> and in NuGet packages.
-        /// </remarks>
-        public ScriptOptions WithDefaultSourceResolution(ImmutableArray<string> searchPaths)
-        {
-            var resolver = new SourceFileResolver(
-                ToImmutableArrayChecked(searchPaths, nameof(searchPaths)),
-                baseDirectory: null);
-
-            return new ScriptOptions(this) { SourceResolver = resolver };
-        }
-
-        /// <summary>
-        /// Creates a new <see cref="ScriptOptions"/> with <see cref="SourceResolver"/> set to the default source resolver for the current platform.
-        /// </summary>
-        /// <param name="searchPaths">Directories to be used by the default resolver when resolving script file names.</param>
-        /// <remarks>
-        /// The default resolver looks up scripts in specified <paramref name="searchPaths"/> and in NuGet packages.
-        /// </remarks>
-        public ScriptOptions WithDefaultSourceResolution(IEnumerable<string> searchPaths) =>
-           WithDefaultSourceResolution(searchPaths.AsImmutableOrEmpty());
-
-        /// <summary>
-        /// Creates a new <see cref="ScriptOptions"/> with <see cref="SourceResolver"/> set to the default source resolver for the current platform.
-        /// </summary>
-        /// <param name="searchPaths">Directories to be used by the default resolver when resolving script file names.</param>
-        /// <remarks>
-        /// The default resolver looks up scripts in specified <paramref name="searchPaths"/> and in NuGet packages.
-        /// </remarks>
-        public ScriptOptions WithDefaultSourceResolution(params string[] searchPaths) =>
-           WithDefaultSourceResolution(searchPaths.AsImmutableOrEmpty());
-
-        /// <summary>
         /// Creates a new <see cref="ScriptOptions"/> with specified <see cref="SourceResolver"/>.
         /// </summary>
-        public ScriptOptions WithCustomSourceResolution(SourceReferenceResolver resolver) =>
+        public ScriptOptions WithSourceResolver(SourceReferenceResolver resolver) =>
             SourceResolver == resolver ? this : new ScriptOptions(this) { SourceResolver = resolver };
```

Comment:
Now that I'm looking at this, it doesn't feel quite right.  I think this should either throw or null out the SourceResolver.  Why would we no-op on null?